parts/linux/cloud-init/artifacts/cgroup-stats.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+set -o errexit
+
+EVENTS_LOGGING_DIR=/var/log/azure/Microsoft.Azure.Extensions.CustomScript/events/
+EVENTS_FILE_NAME=$(date +%s%3N)
+CSE_STARTTIME=$(date)
+CSE_STARTTIME_FORMATTED=$(date +"%F %T.%3N")
+CSE_ENDTIME_FORMATTED=$(date +"%F %T.%3N")
+NODEPOOL_NAME=$(curl -s -H Metadata:true --noproxy "*" "http://169.254.169.254/metadata/instance?api-version=2021-02-01" | jq -r '.compute.osProfile.computerName')
+VM_TYPE=$(kubectl describe node $NODEPOOL_NAME --kubeconfig var/lib/kubelet/kubeconfig | grep "beta.kubernetes.io/instance-type=" | cut -d '=' -f 2)
+
+memory_string=$( jq -n \
+--arg USER_SLICE_MEMORY "$(cat /sys/fs/cgroup/user.slice/memory.current)" \
+--arg AZURE_SLICE_MEMORY "$(cat /sys/fs/cgroup/azure.slice/memory.current)" \
+--arg KUBEPODS_SLICE_MEMORY "$(cat /sys/fs/cgroup/kubepods.slice/memory.current)" \
+--arg SYSTEM_SLICE_MEMORY "$(cat /sys/fs/cgroup/system.slice/memory.current)" \
+--arg CONTAINERD_MEMORY "$(cat /sys/fs/cgroup/system.slice/containerd.service/memory.current)" \
+--arg KUBELET_MEMORY "$(cat /sys/fs/cgroup/system.slice/kubelet.service/memory.current)" \
+--arg ALLOCATABLE_MEMORY "$(cat /sys/fs/cgroup/kubepods.slice/memory.max)" \
+--arg EMPLOYED_MEMORY "$(( $(cat /sys/fs/cgroup/user.slice/memory.current) + $(cat /sys/fs/cgroup/azure.slice/memory.current) + $(cat /sys/fs/cgroup/kubepods.slice/memory.current) + $(cat /sys/fs/cgroup/system.slice/memory.current) ))" \
+--arg CAPACITY_MEMORY "$(kubectl describe node $NODEPOOL_NAME --kubeconfig var/lib/kubelet/kubeconfig | grep "memory:" | head -n 1 | awk '{print $2*1024}' | sed 's/[^0-9]*//g')" \
+'{ UserSliceMemory: $USER_SLICE_MEMORY, AzureSliceMemory: $AZURE_SLICE_MEMORY, KubepodsSliceMemory: $KUBEPODS_SLICE_MEMORY, SystemSliceMemory: $SYSTEM_SLICE_MEMORY, ContainerdMemory: $CONTAINERD_MEMORY, KubeletMemory: $KUBELET_MEMORY, AllocatableMemory: $ALLOCATABLE_MEMORY, EmployedMemory: $EMPLOYED_MEMORY, CapacityMemory: $CAPACITY_MEMORY } | tostring'
+)
+
+pressure_string=$( jq -n \
+--arg MEMORY_PRESSURE "$(cat /sys/fs/cgroup/memory.pressure)" \
```

Comment:
can we do this for:
- system.slice
- azure.slice
- kubepods.slice
- kubelet.service/scope
- containerd.service/scope

not that if we changed the cgroups, you would need to figure them out. e.g.
```
systemctl show containerd -p Slice
Slice=system.slice
```

Reply:
Ok I made it where it finds appropriate slice for kubelet and containerd metrics